lib/libc/gen/psignal.c

```diff
@@ -61,3 +62,91 @@ psiginfo(const siginfo_t *si, const char *s)
 {
 	psignal(si->si_signo, s);
 }
+
+int
+sig2str(int signum, char *str)
+{
+	char tmp[16];
+	char *t, *p;
+	int n;
+
+	if (signum > 0 && signum < sys_nsig) {
+		strcpy(str, sys_signame[signum]);
+		return (0);
+	}
+
+	if (signum < SIGRTMIN || signum > SIGRTMAX)
```

Comment:
This leaves SIGTHR, SIGRTLIB, and [SIGRTLIB + 1, SIGRTMIN - 1] with error, instead of following the requirement
```
66907
If signum is a valid, supported signal number, is either less than SIGRTMIN or greater than
66908
SIGRTMAX, and is not equal to one of the symbolic constants listed in the table of signal
66909
numbers in <signal.h>, the stored string shall uniquely identify the signal number signum in an
66910
unspecified manner.
```

Reply:
> This leaves SIGTHR, SIGRTLIB, and [SIGRTLIB + 1, SIGRTMIN - 1] with error, instead of following the requirement

Will look into this. Beware, though, that `strsignal(SIGTHR)` returns `Unknown signal: 32`.